azavea/temperate
# Add AngularJS bundle to CI pipeline

## Overview
Builds AngularJS bundle and package into an Nginx container for deployment. Also, remove CloudFront CDN, and serve Django static files with Nginx as well.

### Changes:
- Add nginx container
    - Serves angular app from `/`
    - Serves django static files from `/static/`
    - Proxy requests to `/health-check/`, `/accounts` , `/api-token-auth`, `/api-auth`, `/admin`,`api` to django
- Add Angular bundle to nginx in CI
- Add Terraform resources for an ECS service to perform http --> https redirect
- Modify HTTPS ALB target group so that requests go to nginx instead of django
- Remove CloudFront CDN and associated DNS records.

### Demo

Optional. Screenshots, `curl` examples, etc.

### Notes

Optional. Ancillary topics, caveats, alternative strategies that didn't work out, anything else.


## Testing Instructions
* Restart Vagrant VM to make sure Nginx port is exposed
* Run `scripts/update` to build Angular bundle and nginx container
* Run `scripts/server` and visit the app at http://localhost:8000.
* Make sure django endpoints work by hitting http://localhost:8000/api and http://localhost:8000/admin
* See https://plaint.staging.climate.azavea.com

Closes #107 
Closes #118
